mail_cleanup/mail_cleanup.py

```diff
@@ -0,0 +1,144 @@
+
+import logging
+import re
+from osv import fields
+from osv import osv
+from dateutil import relativedelta
+from datetime import datetime
+
+from server_environment import serv_config
+
+_logger = logging.getLogger(__name__)
+
+
+class FetchmailServer(osv.osv):
+    """Incoming POP/IMAP mail server account"""
+    _inherit = 'fetchmail.server'
+
+    def _get_cleanup_conf(self, cursor, uid, ids, name, args, context=None):
+        """
+        Return configuration
+        """
+        res = {}
+        for fetchmail in self.browse(cursor, uid, ids):
+            global_section_name = 'incoming_mail'
+
+            # default vals
+            config_vals = {'cleanup_days': 30,
+                           'cleanup_folder': False}
+            if serv_config.has_section(global_section_name):
+                config_vals.update(serv_config.items(global_section_name))
+
+            custom_section_name = '.'.join((global_section_name,
+                                            fetchmail.name))
+            if serv_config.has_section(custom_section_name):
+                config_vals.update(serv_config.items(custom_section_name))
+
+            if config_vals.get('cleanup_days'):
+                config_vals['cleanup_days'] = \
+                    int(config_vals['cleanup_days'])
+            if config_vals.get('cleanup_folder'):
+                config_vals['cleanup_folder'] = \
+                    config_vals['cleanup_folder']
+            res[fetchmail.id] = config_vals
+        return res
+
+    _columns = {
+        'cleanup_days': fields.function(
+            _get_cleanup_conf,
+            method=True,
+            string='Expiration days',
+            type="integer",
+            multi='outgoing_mail_config',
+            help="Number of days before discarding an e-mail"),
+        'cleanup_folder': fields.function(
+            _get_cleanup_conf,
+            method=True,
+            string='Expiration folder',
+            type="char",
+            multi='outgoing_mail_config',
+            help="Folder where the discarded e-mail will be moved.")
+    }
+
+    def fetch_mail(self, cr, uid, ids, context=None):
+        """ Called before the fetch, in order to clean up
+             right before retrieving emails. """
+        if context is None:
+            context = {}
+        context['fetchmail_cron_running'] = True
+        for server in self.browse(cr, uid, ids, context=context):
+            _logger.info('start cleaning up emails on %s server %s',
+                         server.type, server.name)
+            context.update({'fetchmail_server_id': server.id,
+                            'server_type': server.type})
+            count, failed = 0, 0
+            imap_server = False
+            if server.type == 'imap' and server.cleanup_days > 0:
+                try:
+                    expiration_date = (datetime.now() +
+                                       relativedelta.relativedelta(
```

Comment:
I would extract this big in a helper function. This will save you 12 columns which means fewer awkward line splits.